pytorch_lightning/loops/fit_loop.py

```diff
@@ -175,6 +177,10 @@ def on_advance_start(self) -> None:
         if self.current_epoch != 0 and self.trainer._should_reload_dl_epoch:
             self.trainer.reset_train_dataloader(model)
 
+        if self._dataloader_state_dict:
+            self.trainer.train_dataloader.load_state_dict(self._dataloader_state_dict)
+            self._dataloader_state_dict = {}
```

Comment:
Did you consider moving this inside `reset_train_dataloader`?

Reply:
is that correct? what if we don't want to reset the loader and still just load the state dict? see the if-clause above

for example are in current epoch 0 and want to resume mid epoch